frontend/app/components/WelcomeScreen/hooks/useWelcomeAnimations.ts

```diff
@@ -0,0 +1,261 @@
+      { scale: pulseAnim.value },
+      {
+        rotate: `${interpolate(pulseAnim.value, [1, 1.1], [0, 2])}deg`,
+      },
+    ],
+  }));
+
+  const starsStyle = useAnimatedStyle(() => ({
+    opacity: starsOpacity.value,
+  }));
+
+  const cardFlipStyle = useAnimatedStyle(() => ({
+    transform: [
+      { rotateY: `${interpolate(cardFlipAnim.value, [0, 1], [0, 180])}deg` },
+    ],
+  }));
+
+  const cardBackStyle = useAnimatedStyle(() => ({
+    transform: [
+      { rotateY: `${interpolate(cardFlipAnim.value, [0, 1], [180, 360])}deg` },
+    ],
+  }));
+
+  // Individual sparkle styles
+  const sparkle1Style = useAnimatedStyle(() => ({
+    opacity: sparkle1.value,
+    transform: [{ scale: sparkle1.value }],
+  }));
+
+  const sparkle2Style = useAnimatedStyle(() => ({
+    opacity: sparkle2.value,
+    transform: [{ scale: sparkle2.value }],
+  }));
+
+  const sparkle3Style = useAnimatedStyle(() => ({
+    opacity: sparkle3.value,
+    transform: [{ scale: sparkle3.value }],
+  }));
+
+  const sparkle4Style = useAnimatedStyle(() => ({
+    opacity: sparkle4.value,
+    transform: [{ scale: sparkle4.value }],
+  }));
+
+  const sparkle5Style = useAnimatedStyle(() => ({
+    opacity: sparkle5.value,
+    transform: [{ scale: sparkle5.value }],
+  }));
+
+  const sparkle6Style = useAnimatedStyle(() => ({
+    opacity: sparkle6.value,
+    transform: [{ scale: sparkle6.value }],
+  }));
+
+  // Individual scroll styles
+  const scroll1Style = useAnimatedStyle(() => ({
+    transform: [
+      {
+        translateY: interpolate(scroll1.value, [0, 1], [0, -10]),
+      },
+      { scale: 1.1 },
+    ],
+    opacity: interpolate(scroll1.value, [0, 0.5, 1], [0.7, 1, 0.7]),
+  }));
+
+  const scroll2Style = useAnimatedStyle(() => ({
+    transform: [
+      {
+        translateY: interpolate(scroll2.value, [0, 1], [0, -10]),
+      },
+      { scale: 0.95 },
+    ],
+    opacity: interpolate(scroll2.value, [0, 0.5, 1], [0.7, 1, 0.7]),
+  }));
+
+  const scroll3Style = useAnimatedStyle(() => ({
+    transform: [
+      {
+        translateY: interpolate(scroll3.value, [0, 1], [0, -10]),
+      },
+      { scale: 0.95 },
+    ],
+    opacity: interpolate(scroll3.value, [0, 0.5, 1], [0.7, 1, 0.7]),
+  }));
+
+  // Individual emoji styles
+  const emoji1Style = useAnimatedStyle(() => ({
+    opacity: interpolate(pulseAnim.value, [1, 1.2], [0.7, 1]),
+    transform: [
+      {
+        translateY: interpolate(pulseAnim.value, [1, 1.2], [0, -5]),
+      },
+    ],
+  }));
+
+  const emoji2Style = useAnimatedStyle(() => ({
+    opacity: interpolate(pulseAnim.value, [1, 1.2], [0.7, 1]),
+    transform: [
+      {
+        translateY: interpolate(pulseAnim.value, [1, 1.2], [0, -5]),
+      },
+    ],
+  }));
+
+  const emoji3Style = useAnimatedStyle(() => ({
+    opacity: interpolate(pulseAnim.value, [1, 1.2], [0.7, 1]),
+    transform: [
+      {
+        translateY: interpolate(pulseAnim.value, [1, 1.2], [0, -5]),
+      },
+    ],
+  }));
```

Comment:
It seems some things here are extremely similar or duplicates, can we refactor these into functions or other data sstructures to reduce the duplications and the weird names with numbers?